### PR TITLE
feat: static & ssr

### PR DESCRIPTION
**Pull request by Brentlok**

### CODE REVIEW

File: src/web/utils/unistyle.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@ import type { UnistyleDependency } from '../../specs/NativePlatform'
 import { ColorScheme, Orientation } from '../../specs/types'
 import type { StyleSheet, StyleSheetWithSuperPowers, UnistylesValues } from '../../types/stylesheet'
 import { isUnistylesMq, parseMq } from '../../mq'
-import { UnistylesState } from '../state'
 import { keyInObject, reduceObject } from './common'
+import type { UnistylesBreakpoints } from '../../global'
+import { UnistylesWeb } from '..'
```
Comment (jpudysz) [inline src/web/utils/unistyle.ts:7]: import from index then

File: src/web/utils/unistyle.ts [hunk 2]
```diff
@@ -77,9 +63,18 @@ export const getMediaQuery = (query: string) => {
         ].filter(Boolean).join(' and ')
     }
 
-    const minWidth = UnistylesState.breakpoints && keyInObject(UnistylesState.breakpoints, query) ? UnistylesState.breakpoints[query] : undefined
+    const breakpointValue = UnistylesWeb.runtime.breakpoints[query as keyof UnistylesBreakpoints] ?? 0
+    const nextBreakpoint = allBreakpoints
+            .filter((b): b is keyof UnistylesBreakpoints => b in UnistylesWeb.runtime.breakpoints)
+            .map(b => UnistylesWeb.runtime.breakpoints[b] ?? 1)
```
Comment (jpudysz) [inline src/web/utils/unistyle.ts:69]: what is 1?

File: src/web/variants.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+import { deepMergeObjects, isDefined } from '../utils'
 
 type StylesWithVariants = {
     variants: Record<string, any>,
     compoundVariants?: Array<Record<string, any> & {
         styles: Record<string, any>
     }>
 }
-const hasVariants = <T extends object>(value: [string, T]): value is [string, T & StylesWithVariants] => 'variants' in value[1]
+const hasVariants = <T extends object>(value: [string, T]): value is [string, T & StylesWithVariants] => isDefined(value[1]) && 'variants' in value[1]
```
Comment (jpudysz) [inline src/web/variants.ts:10]: why [1]?

File: src/web/create.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import type { StyleSheetWithSuperPowers, StyleSheet } from '../types/stylesheet'
+import { UnistylesWeb } from '.'
```
Comment (jpudysz) [inline src/web/create.ts:2]: import from index

File: src/web/convert/object/filter.ts [hunk 5]
```diff
@@ -3,8 +3,8 @@ import { hyphenate } from '../../utils'
 import type { Filters } from '../types'
 import { getObjectStyle } from './objectStyle'
 import { normalizeColor, normalizeNumericValue } from '../utils'
-import { UnistylesRuntime } from '../../runtime'
 import { isUnistylesMq } from '../../../mq'
+import { UnistylesWeb } from '../..'
```
Comment (jpudysz) [inline src/web/convert/object/filter.ts:7]: import from index

File: src/server/useServerUnistyles.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import React, { useRef } from 'react'
+import { StyleSheet } from 'react-native'
+import { useServerInsertedHTML } from 'next/navigation'
+import { UnistylesWeb } from '../web'
+
+declare global {
+    interface Window {
+        UNISTYLES_STATE: any
```
Comment (jpudysz) [inline src/server/useServerUnistyles.tsx:8]: if this is global add `__` prefix

File: package.json [hunk 7]
```diff
@@ -88,6 +94,7 @@
     "husky": "9.1.7",
     "jest": "29.7.0",
     "metro-react-native-babel-preset": "0.77.0",
+    "next": "15.1.1",
```
Comment (jpudysz) [inline package.json:97]: why do we need to depend on next?

File: src/components/ScopedTheme.tsx [hunk 8]
```diff
@@ -21,7 +21,7 @@ export const ScopedTheme: React.FunctionComponent<React.PropsWithChildren<ThemeP
     const mappedChildren = [
         <Apply key={name} name={name} />,
         children,
-        <Apply key='dispose' name={previousScopedTheme} />
+        <Apply key='dispose' name={previousScopedTheme as never} />
```
Comment (jpudysz) [inline src/components/ScopedTheme.tsx:24]: as `keyof UnistylesThemes | undefined`

**Review by jpudysz — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (Brentlok) [reply package.json:97]: it's in dev dependencies for `useServerUnistyles` types

**Review by jpudysz — APPROVED** [review]

**Pull request merged**